src/sonic-config-engine/minigraph.py

```diff
@@ -698,7 +715,11 @@ def parse_asic_meta(meta, hname):
                 value = device_property.find(str(QName(ns1, "Value"))).text
                 if name == "SubRole":
                     sub_role = value
-    return sub_role
+                elif name == "SwitchId":
+                    switch_id = value
+                elif name == "InstanceName":
+                    instance_name = value
+    return sub_role, switch_id, instance_name
```

Comment:
We also need **switch_type** and **max_cores** attributes in device metadata (Ref: https://github.com/Azure/SONiC/blob/master/doc/voq/voq_hld.md#231-device_metadata)